mava/wrappers/robot_warehouse.py

```diff
@@ -0,0 +1,227 @@
+        converted_actions_mask: np.ndarray = np.array(actions_mask, dtype="float32")
+
+        return converted_observations, converted_actions_mask
+
+
+class AgentIDWrapper(gym.Wrapper):
+    """Add onehot agent IDs to observation."""
+
+    def __init__(self, env: gym.Env):
+        super().__init__(env)
+
+        self.agent_ids = np.eye(self.env.num_agents)
+        _obs_low, _obs_high, _obs_dtype, _obs_shape = (
+            self.env.observation_space.low[0][0],
+            self.env.observation_space.high[0][0],
+            self.env.observation_space.dtype,
+            self.env.observation_space.shape,
+        )
+        _new_obs_shape = (self.env.num_agents, _obs_shape[1] + self.env.num_agents)
+
+        self.observation_space = Box(
+            low=_obs_low, high=_obs_high, shape=_new_obs_shape, dtype=_obs_dtype
+        )
+
+    def reset(self) -> Tuple[np.ndarray, Dict]:
+        """Reset the environment."""
+        obs, info = self.env.reset()
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, info
+
+    def step(self, action: list) -> Tuple[np.ndarray, float, bool, bool, Dict]:
+        """Step the environment."""
+        obs, reward, terminated, truncated, info = self.env.step(action)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, reward, terminated, truncated, info
+
+
+def make_env_single(
+    map_name: str = "rware-tiny-2ag-v1",
+    team_reward: bool = True,
+    add_agent_id: bool = True,
+) -> Callable:
+    """Create a function which creates a fully configured environment."""
+
+    def thunk() -> gym.Env:
+        """Create an environment."""
+        env = GymWrapper(
+            env=gym.make(map_name),
+            team_reward=team_reward,
+        )
+        if add_agent_id:
+            env = AgentIDWrapper(env)
+        return env
+
+    return thunk
+
+
+def make_env(
+    num_envs: int,
+    config: DictConfig,
+) -> Callable:
+    def thunk() -> gym.vector.VectorEnv:
+        if config.arch.async_envs:
+            envs = gym.vector.AsyncVectorEnv(
+                [
+                    make_env_single(
+                        map_name="rware-tiny-2ag-v1",
+                        team_reward=config.system.use_team_reward,
+                        add_agent_id=config.system.add_agent_id,
+                    )
+                    for _ in range(num_envs)
+                ]
+            )
+        else:
+            envs = gym.vector.SyncVectorEnv(
+                [
+                    make_env_single(
+                        map_name="rware-tiny-2ag-v1",
+                        team_reward=config.system.use_team_reward,
+                        add_agent_id=config.system.add_agent_id,
+                    )
+                    for _ in range(num_envs)
+                ]
+            )
+        envs.num_envs = num_envs
+        envs.is_vector_env = True
+        return envs
+
+    return thunk
```

Comment:
We'd never want to use `SyncVectorEnv` unless debugging and only one of these needs an inner method.

```suggestion
def make_env(
    map_name: str = "rware-tiny-2ag-v1",
    team_reward: bool = True,
    add_agent_id: bool = True,
) -> Callable[[], gym.Env]:
    """Create a function which creates a fully configured environment."""

    def thunk() -> gym.Env:
        env = GymWrapper(
            env=gym.make(map_name),
            team_reward=team_reward,
        )
        if add_agent_id:
            env = AgentIDWrapper(env)

        return env

    return thunk


def make_envs(num_envs: int, config: DictConfig) -> gym.vector.VectorEnv:
    envs = gym.vector.AsyncVectorEnv(
        [
            make_env(
                map_name="rware-tiny-2ag-v1",
                team_reward=config.system.use_team_reward,
                add_agent_id=config.system.add_agent_id,
            )
            for _ in range(num_envs)
        ]
    )

    envs.num_envs = num_envs
    envs.is_vector_env = True
    return envs
```